packages/mds-compliance/api.ts

```diff
@@ -145,23 +145,55 @@ function api(app: express.Express): express.Express {
     }
     /* istanbul ignore next */
     function fail(err: Error): void {
-      log.error('readDeviceIds fail', err.stack || err).then(() => {
+      log.error(err.stack || err).then(() => {
         res.status(500).send('server error')
       })
     }
 
-    const start_date = now() - days(365)
-    const end_date = now() + days(365)
+    let start_date = now() - days(365)
     const { policy_uuid } = req.params
+    const { provider_id } = req.query
+    let { end_date } = req.query
+
+    if (!end_date) {
+      end_date = now() + days(365)
+    }
 
     if (!isUUID(policy_uuid)) {
       res.status(400).send({ err: 'bad_param' })
+    } else if (end_date) {
+      end_date = parseInt(end_date)
+      start_date = end_date - days(365)
+      try {
+        const policies = await db.readPolicies({ policy_id: policy_uuid, start_date, end_date })
```

Comment:
Not a blocker, but I imagine this is going to be a rather slow endpoint with all these reads to the DB. We'll need to change this to hit the cache, or figure out a more efficient way to query the DB, or both.

Reply:
It is pretty slow, yeah. Basically everything in `compliance` is mega slow -- even using cache. There is no way to construct historical state from cache, we _must_ use the db. If you look below, you can see how we compute compliance using cache.

Reply:
Ah, the inability to use the cache makes sense. I don't know much SQL, but there must be a better way to hit the DB.

Reply:
That gigantic SQL query is the best our resident SQL master could come up with for reconstructing historical state, I don't think it's going to get much more efficient than this. This endpoint is likely not going to be used in real-time capacities, and more likely by background tasks for aggregating compliance data.

Reply:
👍